evalml/pipelines/components/transformers/column_selectors.py

```diff
@@ -122,3 +122,52 @@ def transform(self, X, y=None):
             pd.DataFrame: Transformed X.
         """
         return super().transform(X, y)
+
+
+class SelectDtypeColumns(ColumnSelector):
```

Comment:
Regarding the name, I feel like the right thing to do refactor `SelectColumns` to allow for logical type selection - that way there's only one `SelectColumns` transformer - but I don't think that's in scope for this issue.

If we follow my suggestion to also allow selecting by semantic tags, I think `SelectByTypeorTag` is a good alternate name. If not, I think `SelectByLogicalType` is also a good name! I don't like `Dtype` because  that's more pandas terminology and I feel it obscures the fact it's done via woodwork.